test/functional/name_txnqueue.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# Licensed under CC0 (Public domain)
+
+# Test that transaction queue works.
+
+from test_framework.names import NameTestFramework
+from test_framework.util import *
+
+class NameTransactionQueueTest(NameTestFramework):
```

Comment:
One thing I would like to see in this test as well is queueing a transaction that is actually valid right away, and making sure it is properly handled as well (i.e. put directly into the mempool and perhaps even directly removed from the queue, but at least removed right when the next block arrives).

Reply:
OK.